src/store/products/products.selectors.ts

```diff
@@ -0,0 +1,11 @@
+import { createSelector } from "reselect";
+
+import { AppState } from "../rootReducer";
+import { ProductId } from "./products.types";
+
+export const getProducts = (state: AppState) => state.products;
+
+export const getProductById = (id: ProductId) =>
+  createSelector(getProducts, products =>
+    products.find(product => product.product_id === id)
+  );
```

Comment:
I'm not quite sure when this is actually used? As the products are always returned as a list. I see the triple equal has been used to ensure no type conversion will occur :P

Reply:
Lovely, this is a crucial ability, as it allows us to have the database structured in a way that's optimal for data storage, i.e. use relations to reduce redundancy, add constraints etc. Yet we can here view that data in a much more readable form.

Reply:
That's exactly right :)